specifyweb/frontend/js_src/lib/components/Router/ReactLazy.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import type { IR } from '../../utils/types';
+import { LoadingScreen } from '../Molecules/Dialog';
+
+function FakeAsync<PROPS extends IR<unknown>>({
+  Element,
+  props,
+}: {
+  readonly Element: React.FunctionComponent<PROPS>;
+  readonly props: PROPS;
+}) {
+  return (
+    <React.Suspense fallback={<LoadingScreen />}>
+      <Element {...props} />
+    </React.Suspense>
+  );
+}
+
+export function ReactLazy<PROPS extends IR<unknown>>(
```

Comment:
(nitpick) for the future, try to have the main function at the top of the file (ReactLazy), and other variables and functions called by it (FakeAsync) below it in the order they appear in the main function (ReactLazy)

this way code can be read from top to bottom, in and order that is closer to order in which it executes (rather than jumping from bottom to top and back)

Reply:
thanks for reminding that!